lib/managers/window_manager.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <vector>
+
+#include "graphics/libs.h"
+
+typedef size_t WindowID;
+
+class WindowManager {
+   public:
+    static GLFWwindow* get_active_window();
+
+    static void set_active_window(WindowID active_window_id);
+
+    static WindowID construct_window(size_t width, size_t height,
```

Comment:
I think `const char* window_title` should also be an argument of this function.
It would be a bit strange to construct a window with default title, make it active (in order to access it), change the title with a GLFW function (which in of itself is quite bulky), and then restore previously discarded active window.

```C++
WindowID new_window = WIndowManager::construct_window(/* . . . */)
WindowId previous_active = WindowManager::get_active_window();
WindowManager::set_active_window(new_window);
glfwSetWindowTitleOrSomething(/* . . . */);
WindowManager::set_active_window(previous_active);
```
or
```C++
WindowID new_window = WindowManager::construct_window(800, 600, "Da Game");
```